src/wing/CCPACSWingCell.cpp

```diff
@@ -407,246 +444,325 @@ TopoDS_Shape CCPACSWingCell::GetSkinGeometry(TiglCoordinateSystem cs) const
+                eta_lim = 1.;
+            }
 
-    if (m_positioningOuterBorder.GetInputType() == CCPACSWingCellPositionSpanwise::InputType::Rib) {
-        planeShapeOB = GetRibCutGeometry(m_positioningOuterBorder.GetRib());
-    }
-    else if (OBEta1 == 1. && OBEta2 == 1.) {
-        // if the outer border of the cell is the outer border of the Component segment
-        // a cutting plane from the outer border of the WCS is created
-        // this is necessary due to cutting precision
-        BRepAdaptor_Surface surf(wsr.GetOuterFace());
-        gp_Pnt p0 = surf.Value(u05, v0);
-        gp_Pnt pU = surf.Value(u05, v1);
-        gp_Pnt pV = surf.Value(u1, v0);
-        gp_Ax3 ax0UV(p0, gp_Vec(p0, pU) ^ gp_Vec(p0, pV), gp_Vec(p0, pU));
+            if ( fabs(eta_le - eta_lim) < tol && fabs(eta_te - eta_lim) < Precision::Confusion() ) {
+                // if the inner border of the cell is the inner border of the Component segment
+                // a cutting plane from the inner border of the WCS is created
+                // this is necessary due to cutting precision
+                CTiglWingStructureReference wsr(m_parent->GetParent()->GetStructure());
+                BRepAdaptor_Surface surf(wsr.GetInnerFace());
+                gp_Pnt p0 = surf.Value(0.5, 0.0);
+                gp_Pnt pU = surf.Value(0.5, 1.0);
+                gp_Pnt pV = surf.Value(1.0, 0.0);
+                gp_Ax3 ax0UV(p0, gp_Vec(p0, pU) ^ gp_Vec(p0, pV), gp_Vec(p0, pU));
+
+                cuttingShape = BRepBuilderAPI_MakeFace(gp_Pln(ax0UV)).Face();
+            } else {
+                gp_Pln cutPlaneIB   = gp_Pln(border_axis);
+                cuttingShape = BRepBuilderAPI_MakeFace(cutPlaneIB).Face();
+            }
+        }
 
-        planeShapeOB = BRepBuilderAPI_MakeFace(gp_Pln(ax0UV)).Face();
+        result = SplitShape(loftShape, cuttingShape);
     }
 
-    builder.Add(cutPlanes, planeShapeLE);
-    builder.Add(cutPlanes, planeShapeTE);
-    builder.Add(cutPlanes, planeShapeIB);
-    builder.Add(cutPlanes, planeShapeOB);
+    TopoDS_Builder builder;
+    TopoDS_Compound compound;
+    builder.MakeCompound(compound);
 
-    // cut the lower or upper loft with the cutPlanes
-    const TopoDS_Shape result = SplitShape(loftShape, cutPlanes);
     TopTools_IndexedMapOfShape faceMap;
     TopExp::MapShapes(result, TopAbs_FACE, faceMap);
+    for (int f = 1; f <= faceMap.Extent(); f++) {
+        TopoDS_Face loftFace = TopoDS::Face(faceMap(f));
+        gp_Pnt faceCenter = GetCentralFacePoint(loftFace);
+        gp_Vec center_loc = faceCenter.XYZ() - border_axis.Location().XYZ();
+        if ( (center_loc).Dot(border_axis.Direction()) > 0 ) {
+            builder.Add(compound, loftFace);
+        }
+    }
 
-    TopoDS_Compound compound;
-    builder.MakeCompound(compound);
+    return compound;
+}
 
-    bool notFound = true;
+TopoDS_Shape CCPACSWingCell::CutChordwise(GeometryCache& cache,
```

Comment:
CutChordwise duplicates lots of code from CutSpanwise (except u/v is switched). Don't know if it's worth it, but we should think about sharing code between both functions.

Reply:
I don't think it's easy to write a function that works for both spanwise and chordwise borders. But you are right, I guess at least the code block associated to the trimming face can be detached into its own function that can trim in u or v direction.